### PR TITLE
Add FAQ to documentation

### LINKED ISSUE #1165: Improve documentation regarding optional dependencies
Currently there is a small section hidden somewhere in the middle of the manual, mentioning optional dependencies like CGLIB, ByteBuddy and Objenesis. I think they belong into a "Build prerequisites" section rather at the top of the user manual. I also believe it would be possible to better explain which dependency does what, especially with regard to the "either-or" between CGLIB and ByteBuddy. They overlap in functionality. Which one is recommended for which case? Why maintain two ways to mock classes? There must be a unique benefit to each library in specific cases. But which one?

One idea is to organise this as a matrix (table) with columns "tool" (containing name in plain prose and Maven/Gradle coordinates with minimal or suggested version) and "purpose" (containing bullet points explaining what the library does and when to add it to the project).

If you think this is too technical for a user manual, you can link off to a "technical background" document, but I think it belongs in the user manual because every user first needs to set up her Maven or Gradle project before starting to learn about "given - when - then" and even be able to run tests in the first place.

### PR DESCRIPTION
**Pull request by kriegaex**
For now, there are two FAQ items explaining optional dependencies in general and
explaining why there are two class mocking variants ByteBuddy vs. CGLIB.

Closes #1165.

---
This change is [Reviewable](https://reviewable.io/reviews/spockframework/spock/1212)

### CODE REVIEW

**Review by szpak — DISMISSED** [review]

File: docs/faq.adoc [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+== Why are there two options (CGLIB vs ByteBuddy) for mocking classes?
+
+The main reason is backwards compatibility. CGLIB came first historically, ByteBuddy was added later. Both libraries
+basically provide identical features from a Spock user's perspective.
+
+If you are starting with Spock now, we recommend ByteBuddy as our preferred choice.
+
+As of today there are no plans to remove CGLIB anytime soon, because the maintenance effort is relatively low. So you
+may safely continue to use CGLIB if e.g. your tests use it in another context already. If you have no compelling reasons
```
Comment (leonard84) [inline docs/faq.adoc:22]:
```suggestion
CGLIB is in legacy support mode, we will support it as long as the maintenance effort is relatively low. So you
may continue to use CGLIB for now if e.g. your tests use it in another context already. If you have no compelling reasons
```

File: docs/faq.adoc [hunk 2]
```diff
@@ -0,0 +1,24 @@
+= Frequently Asked Questions (FAQ)
+include::include.adoc[]
+
+== What are Spock's optional dependencies used for?
+
+Spock runs fine without optional dependencies, hence the term _optional_. So what are they used for?
+
+  * Objenesis (artifact `org.objenesis:objenesis`) is required for mocking classes without default constructors.
+  * Either CGLIB (artifact `cglib:cglib-nodep`) or ByteBuddy (artifact `net.bytebuddy:byte-buddy`) are required for
```
Comment (leonard84) [inline docs/faq.adoc:9]:
```suggestion
  * Either ByteBuddy (artifact `net.bytebuddy:byte-buddy`) or CGLIB (artifact `cglib:cglib-nodep`) are required for
```

File: docs/faq.adoc [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    mocking classes and not just interfaces.
+
+See also <<interaction_based_testing.adoc#_mocking_classes,Mocking Classes>>.
+
+== Why are there two options (CGLIB vs ByteBuddy) for mocking classes?
```
Comment (leonard84) [inline docs/faq.adoc:14]:
```suggestion
== Why are there two options (ByteBuddy vs CGLIB) for mocking classes?
```

**Review by leonard84 — CHANGES_REQUESTED** [review]
It should also be mentioned, that spock will use ByteBuddy if both libraries are on the classpath.

Also please format according to [one sentence per line](https://asciidoctor.org/docs/asciidoc-recommended-practices/#one-sentence-per-line), I know that not every part of the guide follows that guideline yet, but new documentation should.

**Review by leonard84 — APPROVED** [review]

**Pull request merged**